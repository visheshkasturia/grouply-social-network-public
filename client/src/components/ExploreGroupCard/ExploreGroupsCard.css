.explore-card-container{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: solid 4px black;
  border-radius: 10px;
  margin: 10px;
  height: 180px;
  width: 250px;
  background: #f0e7e6;
}