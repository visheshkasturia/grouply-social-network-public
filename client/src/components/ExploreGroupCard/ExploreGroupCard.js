import React, { useState } from 'react';
import { requestToJoinGroup } from '../../apiCalls/requestToJoinGroup';
import './ExploreGroupsCard.css';
/* eslint-disable */
function ExploreGroupCard({ group, user }) {
  const [status, setStatus] = useState();

  const handleJoinGroup = () => {
    requestToJoinGroup(group._id, user._id).then((data) => {
      setStatus(data)
    })
  }

  return (
    <div className="explore-card-container">
      <h3 className="explore-group-Name">{group.groupName}</h3>
      <div className="explore-group-description">{group.description}</div>
      {/* <div className ="explore-group-description">tags:{group.tags[0]}</div> */}
      {/* <div className ="explore-group-description"> tags: {group.tags.map((t) => (
        <li>{t}</li>
      ))}
      </div> */}
      <button type="button" onClick={handleJoinGroup}> Request to Join </button>
      <div>{status}</div>
    </div>
  )
}

export default ExploreGroupCard
