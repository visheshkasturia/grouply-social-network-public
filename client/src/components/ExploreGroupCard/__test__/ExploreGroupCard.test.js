import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ExploreGroupCard from '../ExploreGroupCard'

const group = {
  groupName: "test",
  description: "desc",
}

it('ExploreGroupCard renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ExploreGroupCard group={group} />, div);
});

it('ExploreGroupCard matches snapshot', () => {
  const tree = renderer.create(<ExploreGroupCard group={group} />).toJSON();
  expect(tree).toMatchSnapshot();
});