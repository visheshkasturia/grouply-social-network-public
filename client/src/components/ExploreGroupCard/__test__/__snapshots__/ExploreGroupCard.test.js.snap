// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreGroupCard matches snapshot 1`] = `
<div
  className="explore-card-container"
>
  <h3
    className="explore-group-Name"
  >
    test
  </h3>
  <div
    className="explore-group-description"
  >
    desc
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
     Request to Join 
  </button>
  <div />
</div>
`;
